src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCFTTest.java

```diff
@@ -1083,88 +1078,150 @@ public void should_search_by_case_ids_and_multiple_work_types_and_return_tasks_f
     }
 
-    private void insertTaskInCftTaskDb(String caseId, String taskId, String taskType) {
-        String warnings = "[{\"warningCode\":\"Code1\", \"warningText\":\"Text1\"}, "
-                          + "{\"warningCode\":\"Code2\", \"warningText\":\"Text2\"}]";
-
-
-        ZonedDateTime createdDate = ZonedDateTime.now();
-        String formattedCreatedDate = CAMUNDA_DATA_TIME_FORMATTER.format(createdDate);
-        ZonedDateTime dueDate = createdDate.plusDays(1);
-        String formattedDueDate = CAMUNDA_DATA_TIME_FORMATTER.format(dueDate);
-
-        InitiateTaskRequest req = new InitiateTaskRequest(INITIATION, asList(
-            new TaskAttribute(TASK_TYPE, taskType),
-            new TaskAttribute(TASK_NAME, "aTaskName"),
-            new TaskAttribute(TASK_CASE_ID, caseId),
-            new TaskAttribute(TASK_TITLE, "A test task"),
-            new TaskAttribute(TASK_CASE_CATEGORY, "Protection"),
-            new TaskAttribute(TASK_ROLE_CATEGORY, "LEGAL_OPERATIONS"),
-            new TaskAttribute(TASK_HAS_WARNINGS, true),
-            new TaskAttribute(TASK_WARNINGS, warnings),
-            new TaskAttribute(TASK_AUTO_ASSIGNED, true),
-            new TaskAttribute(TASK_CREATED, formattedCreatedDate),
-            new TaskAttribute(TASK_DUE_DATE, formattedDueDate)
+    @Test
+    public void should_search_by_role_category_judicial_and_return_tasks_with_role_category_as_judiciary() {
+        String taskType = "reviewAddendumHomeOfficeEvidence";
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds(taskType);
+        String taskId = taskVariables.getTaskId();
+
+        common.setupCFTOrganisationalRoleAssignment(headers);
+
+        common.insertTaskInCftTaskDb(taskVariables, taskType, headers);
+
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(asList(
+            new SearchParameter(ROLE_CATEGORY, SearchOperator.IN,
+                singletonList("JUDICIAL")),
+            new SearchParameter(CASE_ID, SearchOperator.IN,
+                singletonList(taskVariables.getCaseId()))
         ));
 
         Response result = restApiActions.post(
-            TASK_ENDPOINT_BEING_TESTED,
-            taskId,
-            req,
-            authenticationHeaders
+            ENDPOINT_BEING_TESTED,
+            searchTaskRequest,
+            headers
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("tasks.size()", equalTo(1))
+            .body("tasks.jurisdiction", everyItem(is("IA")))
+            .body("tasks.case_id", hasItem(taskVariables.getCaseId()))
+            .body("tasks.id", hasItem(taskId))
+            .body("tasks.role_category", everyItem(is("JUDICIAL")))
+            .body("total_records", equalTo(1));
+    }
+
+    @Test
+    public void should_search_by_any_role_category_and_return_tasks_with_appropriate_role_category() {
+        String taskType = "reviewAddendumEvidence";
```

Comment:
could we extend this further to cater for 2 tasks at least with one have LEGAL_OPERATIONS and another role category. It would give us full verification of the ac. I.e. the filter finds more than one task that fall in the given role ctegories.